packages/react-mutation-mapper/src/store/DefaultMutationMapperStore.ts

```diff
@@ -226,15 +227,32 @@ class DefaultMutationMapperStore implements MutationMapperStore {
         }
     }
 
-    public getAnnotatedMutations(transcriptId: string): Mutation[] {
+    @memoize
```

Comment:
memoize the annotation calls to avoid redundant annotation of the same set of mutations

Reply:
i think we already have a memoize decorator.  what is the "weak" indicate?   also, why not just use computed?

Reply:
Not sure what weak indicates but as far as I can see this is the only memoize decorator we use everywhere else. See for example https://github.com/cBioPortal/cbioportal-frontend/blob/2675e99f1eafdeeb0366ff1d18161f427c0fa3af/src/shared/components/query/CancerStudySelector.tsx#L18 and https://github.com/cBioPortal/cbioportal-frontend/blob/2675e99f1eafdeeb0366ff1d18161f427c0fa3af/src/shared/components/query/QueryStore.ts#L38

The reason this function one is not computed is because we call it in an `invoke` function: https://github.com/cBioPortal/cbioportal-frontend/blob/f206b6dc2d0299db6bb3dedfdf629ae5e7df5c12/packages/react-mutation-mapper/src/store/DefaultMutationMapperStore.ts#L671

Also, there is this dynamic `getMutations` function which makes things trickier.  It might be possible to refactor things to be able to use computed here.